feature_engine/variable_manipulation.py

```diff
@@ -73,7 +73,7 @@ def _find_or_check_numerical_variables(
 
     else:
         # check that user entered variables are of type numerical
-        if any(X[variables].select_dtypes(exclude="number").columns):
+        if len(X[variables].select_dtypes(exclude="number").columns) > 0:
```

Comment:
why change this line? it is performing the same logic and the previous one was more pythonic.

Reply:
> Also fixes the edge case variables = 0 being detected as empty set instead of column named 0 (any replaced by check for non-empty list)

Because of this

let's say you have a dataframe where one of the columns is named __0__
if you call _find_or_check_numerical_variables(X, variables = [0,1]) you want it to check the variables named 0 and 1 and throw an error because 0 is categorical.
However, the any logic will return False:
```
>>> X = pd.DataFrame({0: ['ohf39', '130hfin'], 1: [234, 19889]})
>>> not_numeric = X[[0,1]].select_dtypes(exclude="number").columns
>>> not_numeric
Int64Index([0], dtype='int64')
>>> any(not_numeric)
False
```
and no error will be thrown. While
```
>>> len(not_numeric) > 0
True
```
thus the error will be thrown. Again, this went unnoticed because no test was passing a column named 0 to the "wrong" _find_or_check function

Reply:
wow, crazy. I did not know this. Thank you!